src/gui/DatabaseWidget.cpp

```diff
@@ -2070,12 +2070,13 @@ void DatabaseWidget::processAutoOpen()
                         // Machine name matched an exclusion, don't load this database
                         loadDb = false;
                         break;
+                    } else {
+                        // Not matching an exclusion allows loading on all machines
+                        loadDb = true;
                     }
                 } else if (device.compare(hostName, Qt::CaseInsensitive) == 0) {
+                    // Explicitly named for loading
                     loadDb = true;
```

Comment:
Convoluted logic. Suggest rephrasing to

```c++
bool loadDb = true;
for (auto& device : ifDevice.split(",")) {
    if (device.startsWith("!") && device.mid(1).compare(hostName, Qt::CaseInsensitive) == 0) {
        loadDb = false;
        break;
    } else if (device.compare(hostName, Qt::CaseInsensitive) == 0) {
        loadDb = true;
        break;
    }
}
```

Remove breaks if last match should take precedence instead of first match.

Reply:
I'll think about it a little more. The way this feature was implemented by keepass is asinine.